File: Programmers/카펫.py
def solution(brown, yellow):
    answer = []
    
    s_list = []
    s = brown + yellow
    for i in range(1, s+1):
        if s % i == 0:
            s_list.append(i)
    
    for j in range(len(s_list)//2+1):
        if s_list[-j-1]*2 + s_list[j]*2 - 4 == brown:
            answer.append(s_list[-j-1])
            answer.append(s_list[j])
            break
    
    return answer

File: JavaScript/카펫.js
const getMultiples = (num, multiples) => {
    for (let i = 1; i < num + 1; i++) {
        if (num % i === 0) multiples.push(i);
    }
}

function solution(brown, yellow) {
    const allTiles = brown + yellow, multiples = [];
    getMultiples(allTiles, multiples);
    let left = 0, right = multiples.length - 1;
    while (left <= right) {
        if (multiples[left] + multiples[right] - 2 === brown / 2) return [multiples[right], multiples[left]];
        left += 1;
        right -= 1;
    }
}

//console.log(solution(10, 2));
console.log(solution(8, 1));